zkvm/src/vm.rs

```diff
@@ -626,3 +626,40 @@ where
         .map_err(|_| VMError::R1CSInconsistency)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn issue() {
+        let privkey = Scalar::random(&mut rand::thread_rng());
+        let pubkey = VerificationKey::from_secret(&privkey);
+
+        let predicate = match Predicate::from_witness(PredicateWitness::Key(privkey)) {
+            Err(_) => return assert!(false),
+            Ok(x) => x,
+        };
+
+        // pushdata qty
+        // var
+        // pushdata flv
+        // var
+        // pushdata pred <- pubkey to issue to
+        // issue
+        let mut program = vec![(
+            // TBD: replace with real scalars
+            Instruction::Push(Data::Witness(DataWitness::Scalar(Box::new(Scalar::one())))),
```

Comment:
Probably would be good to do these in one place: 

`impl From<Scalar> for Data`
`impl From<Predicate> for Data`
`impl From<Commitment> for Data`
etc.

Reply:
I've added these conversions here: https://github.com/interstellar/slingshot/pull/86

Reply:
Merged #86 in `main`. Please rebase and you could use these convenience methods.